transport_analysis/tests/test_velocityautocorr.py

```diff
@@ -286,10 +427,146 @@ def test_fft_start_stop_step_all_dims(
+        self,
+        step_vtraj,
+        NSTEP,
+        tdim,
+        tdim_factor,
+        tstart=10,
+        tstop=1000,
+        tstep=10,
+    ):
+        # testing self-diffusivity calculated from the VACF of the
+        # "simple" windowed algorithm on unit velocity trajectory
+        # check that start, stop, step is working correctly
+        v_fft = VACF(step_vtraj.atoms, dim_type=tdim, fft=True)
+        v_fft.run()
+        sd_actual = v_fft.sd(start=tstart, stop=tstop, step=tstep)
+        sd_expected = (
+            integrate.simpson(
+                characteristic_poly(NSTEP, tdim_factor)[tstart:tstop:tstep],
+                range(NSTEP)[tstart:tstop:tstep],
+            )
+            / tdim_factor
+        )
+        # 7705160166.66 (act) agrees with 7705162888.88 (exp) to 6 sig figs
+        assert_approx_equal(sd_actual, sd_expected, significant=6)
+
+    @pytest.mark.parametrize(
+        "tdim, tdim_factor",
+        [
+            ("xyz", 3),
+            ("xy", 2),
+            ("xz", 2),
+            ("yz", 2),
+            ("x", 1),
+            ("y", 1),
+            ("z", 1),
+        ],
+    )
+    def test_fft_sd_odd_step_vtraj_all_dims(
+        self, step_vtraj, NSTEP, tdim, tdim_factor
+    ):
+        # testing self-diffusivity (simpson) calculated from the fft VACF of the
+        # unit velocity trajectory
+        v_fft = VACF(step_vtraj.atoms, dim_type=tdim, fft=True)
+        v_fft.run()
+        sd_actual = v_fft.sd_odd()
+        sd_expected = (
+            integrate.trapezoid(
+                characteristic_poly(NSTEP, tdim_factor), range(NSTEP)
+            )
+            / tdim_factor
+        )
+        # 24307638750.0 (exp) agrees with 24307638888.888885 (act) to 8 sig figs
+        assert_approx_equal(sd_actual, sd_expected, significant=8)
+
+    @pytest.mark.parametrize(
+        "tdim, tdim_factor",
+        [
+            ("xyz", 3),
+            ("xy", 2),
+            ("xz", 2),
+            ("yz", 2),
+            ("x", 1),
+            ("y", 1),
+            ("z", 1),
+        ],
+    )
+    def test_fft_sd_odd_start_stop_step_all_dims(
+        self,
+        step_vtraj,
+        NSTEP,
+        tdim,
+        tdim_factor,
+        tstart=10,
+        tstop=1000,
+        tstep=10,
+    ):
+        # testing self-diffusivity (simpson) calculated from the VACF of the
+        # "simple" windowed algorithm on unit velocity trajectory
+        # check that start, stop, step is working correctly
+        v_fft = VACF(step_vtraj.atoms, dim_type=tdim, fft=True)
+        v_fft.run()
+        sd_actual = v_fft.sd_odd(start=tstart, stop=tstop, step=tstep)
+        sd_expected = (
+            integrate.trapezoid(
+                characteristic_poly(NSTEP, tdim_factor)[tstart:tstop:tstep],
+                range(NSTEP)[tstart:tstop:tstep],
+            )
+            / tdim_factor
+        )
+        # 7705160166.66 (exp) agrees with 7705162888.88 (act) to 6 sig figs
```

Comment:
Is this really the self diffusivity? Seems very high. Try with the equivalent trajectory in positions space with the existing MSD module from MDAnalysis and see what you get.

Reply:
Tests have been added (though they are very rough). Will try to make the separate class next!